plugins/Login/tests/Integration/Security/LoginFromAnotherCountryTest.php

```diff
@@ -0,0 +1,172 @@
+    {
+        parent::setUp();
+
+        Fixture::createWebsite('2010-01-01 05:00:00');
+        Fixture::loadAllTranslations();
+
+        $this->loginCountry = null;
+
+        $this->userModel = new UsersManagerModel();
+        $this->userModel->addUser(
+            self::LOGIN,
+            'a98732d98732',
+            'user1@example.com',
+            '2019-03-03'
+        );
+
+        FakeAccess::$identity = self::LOGIN;
+        FakeAccess::$superUser = false;
+        FakeAccess::$idSitesView = [1];
+
+        // select GeoIP provider
+        LocationProvider::setCurrentProvider(GeoIp2\Php::ID);
+    }
+
+    public function testNoEmailIsSentWhenLoggingInFirstTimeAfterRelease()
+    {
+        // start in France
+        $_SERVER['REMOTE_ADDR'] = self::IP_FRANCE;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEmpty($this->loginCountry);
+    }
+
+    public function testNoEmailIsSentWhenLoggingInFromTheSameCountryAsBefore()
+    {
+        // start in USA
+        $_SERVER['REMOTE_ADDR'] = self::IP_USA;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEmpty($this->loginCountry);
+
+        // continue in USA
+        $_SERVER['REMOTE_ADDR'] = self::IP_USA;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEmpty($this->loginCountry);
+    }
+
+    public function testEmailIsSentWhenLoggingInFromADifferentCountry()
+    {
+        // start in France
+        $_SERVER['REMOTE_ADDR'] = self::IP_FRANCE;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEmpty($this->loginCountry);
+
+        // continue in USA
+        $_SERVER['REMOTE_ADDR'] = self::IP_USA;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEquals(Piwik::translate('Intl_Country_US') ,$this->loginCountry);
+    }
+
+    public function testEmailIsSentWhenLoggingInFromAnUnknownCountry()
+    {
+        // start in France
+        $_SERVER['REMOTE_ADDR'] = self::IP_FRANCE;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEmpty($this->loginCountry);
+
+        // continue in an unknown country
+        $_SERVER['REMOTE_ADDR'] = self::IP_UNKNOWN;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEquals(Piwik::translate('General_Unknown') ,$this->loginCountry);
+
+        // continue in the USA again
+        $_SERVER['REMOTE_ADDR'] = self::IP_USA;
+        Piwik::postEvent('Login.authenticate.processSuccessfulSession.end', [self::LOGIN]);
+        $this->assertEquals(Piwik::translate('Intl_Country_US') ,$this->loginCountry);
+    }
+
+    public function emailIsNotSentWhenGeoIpNotAvailable()
```

Comment:
```suggestion
    public function testEmailIsNotSentWhenGeoIpNotAvailable()
```

Otherwise the test won't run.

Reply:
Good spot, fixed